README.rst

```diff
@@ -572,6 +572,19 @@ This section lists ``pip-tools`` features that are currently deprecated.
   recommended to pass the ``--allow-unsafe`` now to adapt to the upcoming
   change.
 
+A Note on Resolvers
+===================
+
+You can choose from either the legacy or the backtracking resolver.
+The backtracking resolver is recommended, and will become the default
+with the 7.0 release.
+
+Use it now with the ``--resolver=backtracking`` option to ``pip-compile``.
+
+See issue https://github.com/jazzband/pip-tools/issues/1717 for example.
```

Comment:
Why would you think this issue is worth mentioning in README?

Reply:
Eh, I was ambivalent on this.  This is the issue that caused be to make this PR, and it provides some context for why one would want to do this.

Better I suppose would be to include the relevant information in the README directly.  I understand that this is far from the first time this issue came up.

Reply:
I'm not too sure on the details here, but would you say something like this is appropriate:

> The legacy resolver will occasionally fail to resolve dependencies. The backtracking
resolver is more robust, but can take longer to run in general.

Reply:
Sounds fine to me, thanks!